packages/webapp/src/containers/Finances/useTransactions.js

```diff
@@ -0,0 +1,215 @@
+import { mapSalesToRevenueItems, mapTasksToLabourItems } from './util';
+import i18n from '../../locales/i18n';
+import { LABOUR_ITEMS_GROUPING_OPTIONS } from './constants';
+import { getComparator } from '../../util/sort';
+
+const transactionTypeEnum = {
+  expense: 'EXPENSE',
+  labourExpense: 'LABOUR_EXPENSE',
+  revenue: 'REVENUE',
+};
+
+// Polyfill for tests and older browsers
+const groupBy = typeof Object.groupBy === 'function' ? Object.groupBy : lodashGroupBy;
+
+const buildLabourTransactionsFromTasks = ({
+  tasks,
+  taskTypes,
+  users,
+  dateFilter,
+  expenseTypeFilter,
+}) => {
+  const filteredTasks = tasks
+    .map((task) => ({ ...task, date: task.complete_date ?? task.abandon_date }))
+    .filter(
+      (task) =>
+        task.duration &&
+        (!dateFilter ||
+          (moment(task.date).isSameOrAfter(dateFilter.startDate, 'day') &&
+            moment(task.date).isSameOrBefore(dateFilter.endDate, 'day'))) &&
+        // We don't have an actual Labour expense type, but we allow to filter by it in the Expense types filter.
+        (!expenseTypeFilter || expenseTypeFilter?.includes(transactionTypeEnum.labourExpense)),
+    );
+
+  // We only want to show one Labour transaction per day. When expanding the item details on how that transaction was summed up from tasks will be displayed.
+  const groupedTasks = groupBy(filteredTasks, ({ date }) => date);
+  const groupedTransactions = [];
+
+  Object.keys(groupedTasks).forEach((date) => {
+    const labourItems = mapTasksToLabourItems(groupedTasks[date], taskTypes, users);
+    const amount = labourItems[LABOUR_ITEMS_GROUPING_OPTIONS.EMPLOYEE].reduce(
+      (sum, task) => sum + task.labourCost,
+      0,
+    );
+    if (amount > 0) {
+      groupedTransactions.push({
+        date,
+        transactionType: transactionTypeEnum.labourExpense,
+        amount: -amount,
+        items: labourItems,
+      });
+    }
+  });
+
+  return groupedTransactions;
+};
+
+const buildExpenseTransactions = ({ expenses, expenseTypes, dateFilter, expenseTypeFilter }) => {
+  return expenses
+    .filter(
+      (expense) =>
+        (!dateFilter ||
+          (moment(expense.expense_date).isSameOrAfter(dateFilter.startDate, 'day') &&
+            moment(expense.expense_date).isSameOrBefore(dateFilter.endDate, 'day'))) &&
+        (!expenseTypeFilter || expenseTypeFilter?.includes(expense.expense_type_id)) &&
+        expense.value > 0,
+    )
+    .map((expense) => {
+      const expenseType = expenseTypes.find(
+        (expenseType) => expenseType.expense_type_id === expense.expense_type_id,
+      );
+      return {
+        icon: expenseType?.farm_id ? 'CUSTOM' : expenseType?.expense_translation_key,
+        date: expense.expense_date,
+        transactionType: transactionTypeEnum.expense,
+        typeLabel: expenseType?.farm_id
+          ? expenseType?.expense_name
+          : i18n.t(`expense:${expenseType?.expense_translation_key}`),
```

Comment:
For the same bug as here:

- https://github.com/LiteFarmOrg/LiteFarm/pull/2924

```suggestion
          : i18n.t(`expense:${expenseType?.expense_translation_key}.EXPENSE_NAME`),
```